lib/network/kmsAWS/Client.ts

```diff
@@ -23,6 +26,11 @@ interface KMSOptions {
         maxVersion?: TLSVersion;
         key?: Buffer | Buffer[];
     };
+    /**
+     * Do not use AWS KMS Key arn but id (to keep backward compatibility)
+     * Not set or false by default to use arn to be future-proof (multi account, region)
+     */
+    noAwsArn?: boolean;
```

Comment:
Wondering:
- can existing customers use the ARN when they already have KMS-encrypted keys or internal keys?
- if not, how do we ensure they will not run into issues? Can we enforce the correct configuration automatically in such case?

Reply:
- for internal keys, they will never reach this kmsAws client. Existing key or non kmsAws key will be sent to the correct client by cloudserver that parse our own scality arn prefix  (different from the kmsAws client's aws arn) to know the key provider.

- if they already have kmsAws keys (using only the KeyId) it's not a problem, for aws KeyId and KeyArn are interchangeable (except for cross-acount). We should still verify with custom if they do handle KeyArn as AWS does.

- Correct configuration would be to use Arn by default, except if customer's own kmsAws doesn't handle Arn. This case should be written in documentation. The only difference for now is we don't plan to migrate aws KeyId to include the aws KeyArn

Reply:
Can you provide an example of what a keyId looks like with noAwsArn set to false or true.

`arn:scality:kms:ext:aws_kms:custom:key/arn:aws:kms:region:accountId:key/<KEY_ID>` vs 
`arn:scality:kms:ext:aws_kms:custom:key/<KEY_ID>`

Reply:
Fixed https://github.com/scality/Arsenal/pull/2349/files#diff-fb4f257ae2f25bd280160e3cdec22c57fdd9aaa0ee35637e056146ce2b35e6ecR29-R39